packages/@lwc/engine-core/src/shared/format.ts

```diff
@@ -38,3 +45,9 @@ export function getErrorComponentStack(vm: VM): string {
 
     return wcStack.reverse().join('\n\t');
 }
+
+export function addErrorComponentStack(vm: VM, error: any): void {
+    if (!isFrozen(error) && isUndefined(error.wcStack)) {
+        error.wcStack = getErrorComponentStack(vm);
```

Comment:
Instead of checking for `isFrozen`, could you instead create a new Error object from `error` ?

Reply:
I considered this approach but after investigating a bit I realized that I don't understand this domain well enough. For example, is defining the `stack` property all I need to do to clone an error before rethrowing it? Would setting the prototype enough to support `instanceof` checks in case this instance is a subclass of `Error`? Will the approach taken be compatible across our supported browser matrix? And finally, will throwing a new error object result in an increased number of cases on our team, given what we already have to deal with by the simple act of creating pass-through getters in LWC?

Reply:
IMO, we should preserve the original throw object. By recreating a new error, we are losing the original constructor eg. `TypeError` or `LDSInternalError`.

As a side note, today I learned that `Object.isFrozen` returns `false` is the argument is not an Object. So are safe here is userland throws for example a Number or a String.